src/caffe/layers/accuracy_layer.cpp

```diff
@@ -38,6 +38,10 @@ void AccuracyLayer<Dtype>::Reshape(
       << "with integer values in {0, 1, ..., C-1}.";
   vector<int> top_shape(0);  // Accuracy is a scalar; 0 axes.
   top[0]->Reshape(top_shape);
+  if (top.size() > 1) {
+    int dim = bottom[0]->count() / bottom[0]->num();
+    top[1]->Reshape(dim, 1, 1, 1);
```

Comment:
A note: in order to better align with ND-Array blob and to be compatible with accuracy axis, please change these two lines into

    vector<int> top_shape2(1, bottom[0]->shape(label_axis_)); // Per-class accuracy is a vector; 1 axes.
    top[1]->Reshape(top_shape2);

And also reshape your buffer blob after here to have `top_shape2`.